fs/io_zenfs.cc

```diff
@@ -481,25 +542,35 @@ ZonedWritableFile::ZonedWritableFile(ZonedBlockDevice* zbd, bool _buffered,
 
   buffered = _buffered;
   block_sz = zbd->GetBlockSize();
-  buffer_sz = block_sz * 256;
-  buffer_pos = 0;
-
   zoneFile_ = zoneFile;
+  buffer_pos = 0;
+  sparse_buffer = nullptr;
+  buffer = nullptr;
 
   if (buffered) {
-    int ret = posix_memalign((void**)&buffer, sysconf(_SC_PAGESIZE), buffer_sz);
+    size_t sparse_buffer_sz;
+
+    sparse_buffer_sz =
+        1024 * 1024 + block_sz; /* one extra block size for padding */
+    int ret = posix_memalign((void**)&sparse_buffer, sysconf(_SC_PAGESIZE),
+                             sparse_buffer_sz);
+
+    if (ret) sparse_buffer = nullptr;
+
+    assert(sparse_buffer != nullptr);
```

Comment:
Should we not return an error here?

Reply:
Yep, I've created https://github.com/westerndigitalcorporation/zenfs/issues/119 to fix that.